### PR TITLE
Add project dropdown to Update project from database dialog

### LINKED ISSUE #18075: add dropdown with open projects to update project dialog
It would be nice to have an option to select one of the already open projects from a dropdown instead of having to look for the location of the sqlproj.

### PR DESCRIPTION
**Pull request by SakshiS-harma**
This PR fixes #18075.
This PR adds the functionality to select projects from the workspace with the help of a dropdown for target project's location.
If the update from project dialog is opened from Connections context, it populates the source database information and picks the first from the list of projects present in the workspace. 
If the dialog is opened from the Projects context, it populates the source database information from the list of connected databases and pre-selects the project that it was opened from; and populates the dropdown with the workspace projects list.
If no projects are present in the workspace, the dropdown remains empty.
image

### CODE REVIEW

File: extensions/sql-database-projects/src/dialogs/updateProjectFromDatabaseDialog.ts [hunk 1]
```diff
@@ -365,17 +365,23 @@ export class UpdateProjectFromDatabaseDialog {
 	private createProjectLocationRow(view: azdata.ModelView): azdata.FlexContainer {
 		const browseFolderButton: azdata.Component = this.createBrowseFileButton(view);
 
-		const value = this.project ? this.project.projectFilePath : '';
+		let values: string[] = [];
+		this.workspaceProjects.forEach(projectUri => {
+			values.push(projectUri.fsPath as string);
+		});
+
+		const value = this.project ? this.project.projectFilePath : (values[0] ? values[0] : '');
```
Comment (kisantia) [inline extensions/sql-database-projects/src/dialogs/updateProjectFromDatabaseDialog.ts:373]: `values[0] ? values[0] : ''` can be simplified to `values[0] ?? ''`

File: extensions/sql-database-projects/src/dialogs/updateProjectFromDatabaseDialog.ts [hunk 2]
```diff
@@ -365,17 +365,23 @@ export class UpdateProjectFromDatabaseDialog {
 	private createProjectLocationRow(view: azdata.ModelView): azdata.FlexContainer {
 		const browseFolderButton: azdata.Component = this.createBrowseFileButton(view);
 
-		const value = this.project ? this.project.projectFilePath : '';
+		let values: string[] = [];
+		this.workspaceProjects.forEach(projectUri => {
+			values.push(projectUri.fsPath as string);
```
Comment (kisantia) [inline extensions/sql-database-projects/src/dialogs/updateProjectFromDatabaseDialog.ts:370]: is the `as string` needed? The type of `fspath` should be string

**Review by kisantia — APPROVED** [review]

File: extensions/sql-database-projects/src/dialogs/updateProjectFromDatabaseDialog.ts (hunk 2 above)
Comment (SakshiS-harma) [reply extensions/sql-database-projects/src/dialogs/updateProjectFromDatabaseDialog.ts]: Yes, it is string. It wasn't necessary, removed it.

File: extensions/sql-database-projects/src/dialogs/updateProjectFromDatabaseDialog.ts (hunk 1 above)
Comment (SakshiS-harma) [reply extensions/sql-database-projects/src/dialogs/updateProjectFromDatabaseDialog.ts]: Done.

**Pull request merged**